### PR TITLE
Proboscis monkey spec

### PR DESCRIPTION
**Pull request by mephenor**
Added a bit more detail on what exactly needs to be changed for the S3 ID part, because it's hard to keep track of all the parts involved.
Details on the IngestionService are kept on a higher level, specifics will be in the tickets.

### CODE REVIEW

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Looks good

File: 35-proboscis-monkey/technical_specification.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  - `staging_to_permanent` needs source and taget object + bucket IDs now
+  - `permanent_to_outbox` needs source and taget object + bucket IDs now
+
+
+## API Definitions:
+
+### Payload Schemas for Events:
+
+- `FileUploadReceived` should include inbox object ID.
+- `FileUploadValidationSuccess` should include staging object ID
```
Comment (TheByronHimes) [inline 35-proboscis-monkey/technical_specification.md:70]: Do we need to also update FileUploadValidationFailure?

**Comment by dontseyit** [comment]
Epic number 35 is Beluga Whale, the related PR is merged already, you should take the 36.

File: 35-proboscis-monkey/technical_specification.md (hunk 1 above)
Comment (KerstenBreuer) [reply 35-proboscis-monkey/technical_specification.md:70]: Maybe this is also a good idea.

File: 35-proboscis-monkey/technical_specification.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- `FileInternallyRegistered` should include permanent storage object ID
```
[the remark this answers is not in the document]
Comment (mephenor) [reply 35-proboscis-monkey/technical_specification.md:71]: Line 74 should cover this: All these events also need to have the corresponding bucket ID included.

File: 35-proboscis-monkey/technical_specification.md (hunk 1 above)
Comment (mephenor) [reply 35-proboscis-monkey/technical_specification.md:70]: Is there anything that needs changes in FileUploadValidationFailure?
i.e. do we need some bucket and object id in there?

File: 35-proboscis-monkey/technical_specification.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+The second task is the implementation of an ingestion service that processes metadata files produced by the file upload script.
+
+### Outline:
+
+#### Ingestion Service:
+
+The ingestion service will provide a RESTful endpoint to process the output files of the upload script, one at a time.
+This endpoint expects the assymetrically encrypted output metadata as body and an authentication token hash in the header.
+To simplify the process, the public key of the encryption keypair is provided as a configuration option to the service, as is the token hash for authentication.
+The actual token from which the hash is derived and the private key remain with the submitter.
```
[the remark this answers is not in the document]
Comment (mephenor) [reply 35-proboscis-monkey/technical_specification.md:17]: Right, somehow mixed that up

File: 35-proboscis-monkey/technical_specification.md (hunk 1 above)
Comment (TheByronHimes) [reply 35-proboscis-monkey/technical_specification.md]: I guess there wouldn't be a bucket or object to include. Nevermind!

**Review by TheByronHimes — APPROVED** [review]

**Pull request merged**